Cesium.CodeGen/Ir/Expressions/BinaryOperators/ComparisonBinaryOperatorExpression.cs

```diff
@@ -41,10 +41,29 @@ public ComparisonBinaryOperatorExpression(Ast.ComparisonBinaryOperatorExpression
 
     public override void EmitTo(IDeclarationScope scope)
     {
-        // TODO[#160]: check if operand types are compatible?
+        var leftType = Left.GetExpressionType(scope);
+        var rightType = Right.GetExpressionType(scope);
+
+        if ((!scope.TypeSystem.IsNumeric(leftType) && !leftType.IsPointer)
+            || (!scope.TypeSystem.IsNumeric(rightType) && !rightType.IsPointer))
+            throw new InvalidOperationException($"Unable to compare {leftType} to {rightType}");
+
+        if (!scope.TypeSystem.IsBool(leftType) && scope.TypeSystem.IsBool(rightType))
+        {
+            var commonType = scope.TypeSystem.GetCommonNumericType(leftType, rightType);
+
+            Left.EmitTo(scope);
+            EmitConversion(scope, leftType, commonType);
+
+            Right.EmitTo(scope);
+            EmitConversion(scope, rightType, commonType);
```

Comment:
Why isn't this work for bool?

Reply:
Because in current version, there is no conversion defined for bool in EmitConversion

Reply:
Well, let's define it? Such checks in generic code are not very good.